lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/NodeDAOImpl.java

```diff
@@ -1033,10 +1067,9 @@ public boolean doesNodeRevisionExist(String nodeId, Long revNumber) {
 	}
 
 	@Override
-	public EntityHeader getEntityHeader(String nodeId, Long versionNumber) throws DatastoreException, NotFoundException {
+	public EntityHeader getEntityHeader(String nodeId) throws DatastoreException, NotFoundException {
 		Reference ref = new Reference();
 		ref.setTargetId(nodeId);
-		ref.setTargetVersionNumber(versionNumber);
```

Comment:
At line 1073 below if(ref.getTargetVersionNumber() != null) block should probably be removed together with the EntityHeader clone = SerializationUtils.cloneJSONEntity(original); one line above.

Reply:
Verified that there is no path from the API to call this method with a non-null version.